scripts/check-test-complexity.sh

```
@@ -0,0 +1,99 @@
+# Check 6: No complex mocking frameworks (exclude e2e/helpers directory)
+MOCK_COUNT=$(find tests -name "*.test.js" -o -name "setup.js" -o -name "helpers.js" -o -name "vitest.config.js" | xargs grep -h "jest\.mock\|sinon\|proxyquire\|mockery" 2>/dev/null | wc -l || echo "0")
+if [ "$MOCK_COUNT" -gt 0 ]; then
+  echo "❌ Found complex mocking frameworks ($MOCK_COUNT occurrences)"
+  COMPLEXITY_ISSUES=$((COMPLEXITY_ISSUES + 1))
+else
+  echo "✅ No complex mocking frameworks"
+fi
+
+# Check 7: No test builders or factories (exclude e2e/helpers directory which contains utilities)
+# Only scan for actual class-based builders/factories, not simple helper functions
+BUILDER_COUNT=$(find tests -name "*.test.js" | xargs grep -h "class.*Builder\|class.*Factory\|extends.*Builder\|extends.*Factory" 2>/dev/null | wc -l || echo "0")
```

Shellcheck (0.10.0)

[warning] Use 'find .. -print0 | xargs -0 ..' or 'find .. -exec .. +' to allow non-alphanumeric filenames.

(SC2038)